cdci_osa_plugin/osa_dataserve_dispatcher.py

```diff
@@ -646,3 +649,32 @@ def get_col_data(t, n, default):
 
         return cat
 
+
+class OsaQuery(ProductQuery): 
+    """
+    base class for all osa
+    """
+    
+    def check_query_roles(self, provided_roles: List[str], par_dic: dict):
+        # TODO: there is this get_par_by_name there, to check if it can be made work
+        
+        max_pointings = int(par_dic.get('max_pointings', 50))
+        scw_list = par_dic['scw_list'] 
```

Comment:
Should we also check that `scw_list` is provided within the query or we assume it will always be there in the case of OsaQuery?

Reply:
it should be always there else it should fail. In principle, it is better if parameters are defined as obligatory at other level.